frost-core/src/frost.rs

```diff
@@ -99,46 +99,28 @@ where
 
 /// Generates the lagrange coefficient for the i'th participant.
 fn derive_lagrange_coeff<C: Ciphersuite>(
-    signer_id: u16,
+    signer_id: &Identifier<C>,
     signing_package: &SigningPackage<C>,
 ) -> Result<<<C::Group as Group>::Field as Field>::Scalar, &'static str> {
-    // This should fail and panic if signer_id_scalar is 0 in the scalar field.
-    let signer_id_scalar = Identifier::<C>::try_from(signer_id).unwrap();
+    let signer_id_scalar = signer_id.to_scalar();
 
     let zero = <<C::Group as Group>::Field as Field>::zero();
 
-    // TODO: This is redundant
-    if signer_id_scalar.0 == zero {
-        return Err("Invalid parameters");
-    }
-
-    if signing_package
-        .signing_commitments()
-        .iter()
-        .any(|commitment| {
-            let commitment_id_scalar = Identifier::<C>::try_from(commitment.index).unwrap();
-
-            *commitment_id_scalar == zero
-        })
-    {
-        return Err("Invalid parameters");
-    }
```

Comment:
?

Reply:
`Identifier` is guaranteed to be non-zero per construction, so no need to check it. We can keep the check for security in depth though.

Reply:
...but that reminded me to remove `pub(crate)` from the underlying index so that we have more assurance that it won't be constructed with zero, done in dc61839

Reply:
Nice